pkg/api/team.go

```diff
@@ -23,7 +23,7 @@ func (hs *HTTPServer) CreateTeam(c *m.ReqContext, cmd m.CreateTeamCommand) Respo
 		return Error(500, "Failed to create Team", err)
 	}
 
-	if c.OrgRole == m.ROLE_EDITOR && hs.Cfg.EditorsCanAdmin {
+	if c.OrgRole == m.ROLE_EDITOR && hs.Cfg.EditorsCanAdmin && !c.SignedInUser.IsAnonymous {
```

Comment:
> 1. What happens if admin user calls this API? Not important right now as it was here before but looks weird to me that this only adds Editor users not admin ones. Does admin already have enough privileges that he do not need to be added into the team?

I was wondering also about this. I will try to find out.

> 2. I am not sure I like the usage of `IsAnonymous` for this. The user is kinda weird in the sense that it will have `IsSignedIn = true` and `IsAnonymous = true` which does not seem logical to me in the context of what those attributes represent. Also `IsAnonymous` is used in other places so not 100% sure what switching it for API key users will do. What about having something like `SignedInUser.IsRealUser()` that will return false if `id = 0`? Maybe that would better represent the state of the user here (real anonymous users, who did not log in, also have `id = 0` so would make sense also for them).

You are right; `IsRealUser` is better option.

> 3. In the issue there is mentioned an error log for this case, I think it would make sense to add it (or maybe more appropriate here would be a warning?)

I completely forgot about this. I will add a log message.

Reply:
1. Yes admins already have access to all teams